compiler/vm/vm.nim

```diff
@@ -572,21 +590,51 @@ template takeAddress(reg, source) =
   when defined(gcDestructors):
     GC_ref source
 
-proc rawExecute(c: var TCtx, start: int, tos: PStackFrame): TFullReg =
+proc rawExecute(c: var TCtx, start: int, sframe: StackFrameIndex): TFullReg =
+  assert sframe == c.sframes.high
+
   var pc = start
-  var tos = tos
+  var tos = sframe
   # Used to keep track of where the execution is resumed.
   var savedPC = -1
-  var savedFrame: PStackFrame
+  var savedFrame: StackFrameIndex
   when defined(gcArc) or defined(gcOrc):
-    template updateRegsAlias = discard
-    template regs: untyped = tos.slots
+    # TODO: does shallowCopy for seqs still work under --gc:arc? 
```

Comment:
It doesn't, unfortunately. `shallowCopy` is gone for `seq` and `strings` in new runtime.